chainstate/types/src/block_status.rs

```diff
@@ -13,29 +13,54 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use bitflags::bitflags;
+use derive_more::Display;
 use enum_iterator::Sequence;
 use serialization::{Decode, Encode};
 
+// FIXME: which repr should I specify for the enums?
+// (Also, can changing the repr in the future lead to incompatibilities with the existing DB?).
+
 /// Block validation steps are always performed in the same order, which is represented by this enum.
-#[derive(Copy, Clone, PartialEq, Eq, Debug, Sequence, Decode, Encode)]
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Debug, Display, Sequence, Decode, Encode)]
 pub enum BlockValidationStage {
     Unchecked,
-    ParentOk,
     CheckBlockOk,
     FullyChecked,
 }
 
+// FIXME: note that I'm using bitflags 1.x here. In 2.x they changed something and now Encode/Decode
+// can't simply be derived for the struct. Is it OK to continue using bitflags 1.x?
+// As a sidenote, I've also looked into several other "bit-flag" crates but they all seem to have
+// their own drawbacks.
+// The alternative is to follow the existing approach of Service/Services in
+// "p2p/src/net/types/services.rs" or UtxoType/UtxoTypes in "wallet/types/src/utxo_types.rs",
+// but I'd then try to generalize it first to avoid code duplication.
```

Comment:
There's a slightly newer version of `[parity-scale-codec](https://github.com/paritytech/parity-scale-codec)` available (3.6.3 while we use 3.5.0). Worth checking if that solves the problem. However, I went through the changelog and there was nothing obvious to indicate that would be the case.

Reply:
I'm happy using `bitflags` since it's fairly widely used. But wouldn't prefer the 1.x version constraint.

Reply:
Updating parity-scale-codec didn't help. Also, as I can see, they never had any specific support for bitflags, it's just that with bitflags 1.x the generated struct contains a simple numeric field, which is already serializable, and in bitflags 2.x they wrap it in some private intermediate type.
So I guess I'll just get rid of bitflags here.